stories/__tests__/SQFormCheckboxGroup.stories.test.js

```diff
@@ -96,55 +96,97 @@ describe('SQFormCheckboxGroup Tests', () => {
     it('should check all checkboxes when select all checkbox checked', () => {
       render(<SQFormCheckboxGroup size="auto" shouldUseSelectAll />);
 
+      const checkboxes = screen.getAllByRole('checkbox');
+
+      const filteredCheckboxes = checkboxes.filter(
+        checkbox => checkbox.checked
+      );
+      expect(filteredCheckboxes).toEqual([]);
+      expect(filteredCheckboxes).toHaveLength(0);
+
       const selectAllCheckbox = screen.getByLabelText('All');
-      const cb1 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[0].label);
-      const cb2 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[1].label);
-      const cb3 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[2].label);
-      const cb4 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[3].label);
-      const cb5 = screen.getByLabelText(SHOPPING_LIST_OPTIONS[4].label);
-      expect(selectAllCheckbox).not.toBeChecked();
-      expect(cb1).not.toBeChecked();
-      expect(cb2).not.toBeChecked();
-      expect(cb3).not.toBeChecked();
-      expect(cb4).not.toBeChecked();
-      expect(cb5).not.toBeChecked();
 
       userEvent.click(selectAllCheckbox);
 
-      expect(selectAllCheckbox).toBeChecked();
-      expect(cb1).toBeChecked();
-      expect(cb2).toBeChecked();
-      expect(cb3).toBeChecked();
-      expect(cb4).toBeChecked();
-      expect(cb5).toBeChecked();
+      const checkedCheckboxes = checkboxes.filter(checkbox => checkbox.checked);
+      expect(checkedCheckboxes).toEqual(
+        expect.arrayContaining([selectAllCheckbox])
+      );
+      expect(checkedCheckboxes).toHaveLength(6);
```

Comment:
Looks good!
Nitpick: for the length it may be more future proof to do `SHOPPING_LIST_OPTIONS.length + 1`, just in case someone later on adds or removes options from that array